src/main/java/uk/ac/ebi/ampt2d/metadata/exceptionhandling/ExceptionHandlers.java

```diff
@@ -33,9 +34,18 @@
         return new ResponseEntity(new ErrorMessage(HttpStatus.BAD_REQUEST, ex, ex.getMessage()), HttpStatus.BAD_REQUEST);
     }
 
+    @ExceptionHandler(value = ConstraintViolationException.class)
+    public ResponseEntity<ErrorMessage> handleConstraintViolationException(ConstraintViolationException ex) {
+        if (ex.getErrorCode() == 23505) { // 23505 is a constraint violation error Code for duplicate id or index
```

Comment:
😨 Don't they provide a nicer way to check this?

Reply:
nope all the constraint violations of the database will be returned by this exception only, it can't be differentiated other than error codes i guess. However all other constraint checks are checked before hand by javaxValidations. Despite that just for sure duplicate key constraint violation i added this checking.